File: jiwon/050 [1차] 비밀지도.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    string anan = "";
    vector<int> a1;
    vector<int> a2;
    
    for(int k=0;k<n;k++){
        for(int i=0;i<n;i++){
            if(arr1[k] % 2 == 1){
                a1.push_back(1);
            } else{
                a1.push_back(0);
            }
            arr1[k] /= 2;
        }
 
        reverse(a1.begin(), a1.end());
        
        for(int i=0;i<n;i++){
            if(arr2[k] % 2 == 1){
                a2.push_back(1);
            } else{
                a2.push_back(0);
            }
            arr2[k] /= 2;
        }
        
        reverse(a2.begin(), a2.end());
        
        for(int i=0;i<n;i++){
            if(a1[i] == 1 || a2[i] == 1){
                anan += '#';
            } else{
                anan += ' ';
            }
        }
        
        answer.push_back(anan);
        
        anan = "";
        a1.clear();
        a2.clear();
    }
        
    return answer;
}

File: lesunny/050_[1차]비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        int[] binaryArr1 = new int[n];
        int[] binaryArr2 = new int[n];
        
        for(int i=0; i<n; i++){
            //arr1 입력
            for(int j=n-1; j>=0; j--){
                if(arr1[i]>0){
                    binaryArr1[j] = arr1[i]%2;
                    arr1[i] /= 2;
                }else
                    binaryArr1[j] = 0;
            }
            
            //arr2 입력
            for(int j=n-1; j>=0; j--){
                if(arr2[i]>0){
                    binaryArr2[j] = arr2[i]%2;
                    arr2[i] /= 2;
                }else
                    binaryArr2[j] = 0;
            }
            
            StringBuffer sb = new StringBuffer();
            
            for(int k=0; k<n; k++){
                if(binaryArr1[k]==1 | binaryArr2[k]==1){
                    sb.append("#");
                }else{
                    sb.append(" ");
                }
            }
            //System.out.println(sb);
            answer[i] = sb.toString();
        }
        return answer;
    }
}
